modules/fedimint-ln-client/src/receive.rs

```diff
@@ -39,12 +39,11 @@ const RETRY_DELAY: Duration = Duration::from_secs(1);
 /// ```
 #[derive(Debug, Clone, Eq, PartialEq, Hash, Decodable, Encodable)]
 pub enum LightningReceiveStates {
-    SubmittedOfferV0(LightningReceiveSubmittedOfferV0),
+    SubmittedOffer(LightningReceiveSubmittedOffer),
```

Comment:
This cleanup requires another migration since currently the `SubmittedOffer` variant has index 5 when encoded, while it is switched back to 0 in this PR, which was the state in the previous DB version.

Reply:
I originally had it like then but ended up squashing since there were some refactors in the later commits that didn't fit. I'll try to preserve the history next time

Reply:
Fixed the commit history! Hopefully it is easier to review now.

I also added the new migration and a test for it.